test_package.go

```diff
@@ -19,15 +19,16 @@ import (
 
 // Deployment provides a way for tests to interact with a set of homeservers.
 type Deployment interface {
-	// Client returns a CSAPI client targeting the given hsName, using the access token for the given userID.
-	// Fails the test if the hsName is not found. Returns an unauthenticated client if userID is "", fails the test
-	// if the userID is otherwise not found.
-	Client(t *testing.T, serverName, userID string) *client.CSAPI
+	// UnauthenticatedClient returns a blank CSAPI client.
+	UnauthenticatedClient(t *testing.T, serverName string) *client.CSAPI
 	// Register a new user on the given server.
 	Register(t *testing.T, hsName string, opts helpers.RegistrationOpts) *client.CSAPI
```

Comment:
I know it's not part of this PR, but looking at the usage, why keep the `hsName` here, instead of using an index?

Also, since most of the times, the registration options are not set, can't we have an API with variadic args? So that `d.Register(t, "hs1")` works, and you can still do `d.Register(t, "hs1", WithDeviceID("foo"), WithLocalPartSuffix("suffix"))`

Reply:
I chose to keep `hsName` mostly to try to keep things not too dissimilar to the previous API shape. It'll likely change again in another PR.

Agreed variadic functional args would be clearer, I'll do it on another PR.